repo.go

```diff
@@ -695,6 +696,41 @@ func (r *Repo) AddTargets(paths []string, custom json.RawMessage) error {
 	return r.AddTargetsWithExpires(paths, custom, data.DefaultExpires("targets"))
 }
 
+func (r *Repo) AddDigestTargets(digest string, length int64, custom json.RawMessage, path string) error {
+	expires := data.DefaultExpires("targets")
+
+	t, err := r.targets()
+	if err != nil {
+		return err
+	}
+	splitDigest := strings.Split(digest, ":")
+	if len(splitDigest) != 2 {
+		return fmt.Errorf("incorrect format for digest: %s", digest)
+	}
+	hash := splitDigest[1]
+	hashAlg := splitDigest[0]
```

Comment:
The `sha256:ABC123` format is pretty specific to OCI refs, right? What do you think about passing the hash algorithm and hash as separate arguments, instead of requiring a special format for `digest`?